### PR TITLE
Expand ppl command

### PR DESCRIPTION
**Pull request by YANG-DB**
### Description
support `expand` command

### Related Issues
#657

### Check List
- [x] Updated documentation (docs/ppl-lang/README.md)
- [x] Implemented unit tests
- [x] Implemented tests for combination with other commands
- [x] New added source code should include a copyright header
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 1]
```diff
@@ -459,13 +461,34 @@ public LogicalPlan visitFlatten(Flatten flatten, CatalystPlanContext context) {
...
+            // Create an UnresolvedStar for all-fields projection
+            context.getNamedParseExpressions().push(UnresolvedStar$.MODULE$.apply(Option.<Seq<String>>empty()));
+        }
+        Expression field = visitExpression(node.getField(), context);
+        Optional<Expression> alias = node.getAlias().map(aliasNode -> visitExpression(aliasNode, context));
+        context.retainAllNamedParseExpressions(p -> (NamedExpression) p);
+        Explode explodeGenerator = new Explode(field);
+        scala.collection.mutable.Seq seq = alias.isEmpty() ? seq() : seq(alias.get());
+        if(alias.isEmpty())
+            return context.apply(p -> new Generate(new GeneratorOuter(explodeGenerator), seq(), true, (Option) None$.MODULE$, seq, p));
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:484]: Here we do not need the `GeneratorOuter`. How about simplify to
```
return context.apply(p -> new Generate(explodeGenerator, seq(), false, (Option) None$.MODULE$, seq, p));
```

Note. the third parameter `outer` should be `false` here.

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java [hunk 2]
```diff
@@ -459,13 +461,34 @@ public LogicalPlan visitFlatten(Flatten flatten, CatalystPlanContext context) {
...
+        else {
+            //in case an alias does appear - remove the original field from the returning columns
+            context.apply(p -> new Generate(new GeneratorOuter(explodeGenerator), seq(), true, (Option) None$.MODULE$, seq, p));
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java:487]: ditto

File: docs/ppl-lang/PPL-Example-Commands.md [hunk 3]
```diff
@@ -437,8 +437,28 @@ Assumptions: `a`, `b` are fields of table outer, `c`, `d` are fields of table in
...
+   -  `source=accounts | fillnull fields request_path='/not_found', timestamp='*'`
+    - `source=accounts | fillnull using field1=101`
+    - `source=accounts | fillnull using field1=concat(field2, field3), field4=2*pi()*field5`
+    - `source=accounts | fillnull using field1=concat(field2, field3), field4=2*pi()*field5, field6 = 'N/A'`
+```
+
+#### **expand**
+[See additional command details](ppl-expand-command.md)
+```sql
+   -  `source=  table | expand field_with_array as array_list`
```
[the remark this answers is not in the document]
Comment (LantaoJin) [reply docs/ppl-lang/PPL-Example-Commands.md:454]: And add a similar test case in IT.

File: ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java (hunk 1 above)
Comment (YANG-DB) [reply ppl-spark-integration/src/main/java/org/opensearch/sql/ppl/CatalystQueryPlanVisitor.java]: @LantaoJin the issue here is that in the case where row is `null` not using `GeneratorOuter` will not output the row as shown here:

```sql
 ( 1, array(STRUCT("1_one", 1), STRUCT(null, 11), STRUCT("1_three", null)) ),
 ( 2, array(STRUCT("2_Monday", 2), null) ),
 ( 3, array(STRUCT("3_third", 3), STRUCT("3_4th", 4)) ),
 ( 4, null )
```
For the query:
`source = $multiValueTable | expand multi_value AS exploded_multi_value | fields exploded_multi_value`

**Review by LantaoJin — APPROVED** [review]

**Pull request merged**